client/client-multi/client-cpp/kaa/NetworkAddrChangeDetector/NetworkAddrChangeDetector.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _NETWORK_ADDR_CHANGE_DETECTOR_
+#define _NETWORK_ADDR_CHANGE_DETECTOR_
+
+#include <string>
+#include <stdexcept>
+#include <functional>
+
+namespace kaa {
+
+/**
+ * The class is responsible for observation ip address changes and notifying of the subscriber
+ * if the ip address has been changed.
+ */
+class NetworkAddrChangeDetector {
```

Comment:
This class does nothing. I mean... literally _nothing_.

I would just store a `std::function<void(const std::string &newAddrName)> on_reconnect;` in the client context (with default doing nothing) and call it whenever client reconnect.

The logic in `check()` is too simple to deserve a class. Furthermore, not checking `currentIp != ip` allows more usage by the user (e.g., check if we reconnect at all).